### PR TITLE
MULE-10946 Remove Event.isTransacted() and Event.refreshSync()

### PR DESCRIPTION
**Pull request by dfeist**
PENDING: https://muleesb.ci.cloudbees.com/job/Mule-4.x-CE-DF/195/

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/message/DefaultEventBuilder.java [hunk 1]
```diff
@@ -225,31 +216,21 @@ public Builder disableNotifications() {
...
-  @Override
   public Event build() {
     if (originalEvent != null && !modified) {
       return originalEvent;
     } else {
-      return new EventImplementation(context, message, flowVariables, exchangePattern, flow, session, transacted,
-                                     synchronous == null ? (resolveEventSynchronicity() && replyToHandler == null)
-                                         : synchronous,
+      return new EventImplementation(context, message, flowVariables, exchangePattern, flow, session,
+                                     synchronous == null ? resolveEventSynchronicity() : synchronous,
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/message/DefaultEventBuilder.java]: does removing the replyToHandler check keep the same functionality? doesn't it affect transports?

File: core/src/main/java/org/mule/runtime/core/transaction/TransactionCoordination.java [hunk 2]
```diff
@@ -225,4 +225,13 @@ public void restoreIsolatedTransaction() {
       transactions.set((Transaction) isolatedTransactions.get().pop());
     }
   }
+
+  /**
+   * Determine is there is an active transaction associated with the current thread.
+   *
+   * @return true if there is an active transaction, false otherwise.
+   */
+  public static boolean isTransactionActive() {
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/transaction/TransactionCoordination.java]: should we add a TODO to remove the thread/tx association?

File: tests/integration/src/test/java/org/mule/test/construct/FlowConfigurationFunctionalTestCase.java [hunk 3]
```diff
@@ -393,7 +394,7 @@ public void testAsyncRequestResponseEndpoint() throws Exception {
 
   @Test
   public void testAsyncTransactionalEndpoint() throws Exception {
-    Exception e = flowRunner("async-tx").withPayload("0").transactionally(ACTION_NONE, new TestTransactionFactory())
+    Exception e = flowRunner("async-tx").withPayload("0").transactionally(ACTION_ALWAYS_BEGIN, new TestTransactionFactory())
```
Comment (elrodro83) [inline tests/integration/src/test/java/org/mule/test/construct/FlowConfigurationFunctionalTestCase.java]: why this change?

File: tests/integration/src/test/java/org/mule/test/construct/FlowDefaultProcessingStrategyTestCase.java [hunk 4]
```diff
@@ -37,43 +38,34 @@ public void requestResponse() throws Exception {
...
     flowRunner(FLOW_NAME).withPayload(TEST_PAYLOAD).asynchronously().run();
     InternalMessage message = muleContext.getClient().request("test://out", RECEIVE_TIMEOUT).getRight().get();
-    assertThat(message.getOutboundProperty(PROCESSOR_THREAD), is(not(Thread.currentThread().getName())));
+    assertThat(message.getOutboundProperty(PROCESSOR_THREAD), is(not(currentThread().getName())));
   }
 
   @Test
   public void requestResponseTransacted() throws Exception {
-    flowRunner("Flow").withPayload(TEST_PAYLOAD).transactionally(TransactionConfigEnum.ACTION_NONE, new TestTransactionFactory())
+    flowRunner("Flow").withPayload(TEST_PAYLOAD).transactionally(ACTION_ALWAYS_BEGIN, new TestTransactionFactory())
```
Comment (elrodro83) [inline tests/integration/src/test/java/org/mule/test/construct/FlowDefaultProcessingStrategyTestCase.java]: why this change?

File: tests/integration/src/test/java/org/mule/test/construct/FlowDefaultProcessingStrategyTestCase.java [hunk 5]
```diff
@@ -37,43 +38,34 @@ public void requestResponse() throws Exception {
...
 
     InternalMessage message = muleContext.getClient().request("test://out", RECEIVE_TIMEOUT).getRight().get();
-    assertThat(message.getOutboundProperty(PROCESSOR_THREAD), is(Thread.currentThread().getName()));
+    assertThat(message.getOutboundProperty(PROCESSOR_THREAD), is(currentThread().getName()));
   }
 
   @Test
   public void oneWayTransacted() throws Exception {
-    flowRunner("Flow").withPayload(TEST_PAYLOAD).transactionally(TransactionConfigEnum.ACTION_NONE, new TestTransactionFactory())
+    flowRunner("Flow").withPayload(TEST_PAYLOAD).transactionally(ACTION_ALWAYS_BEGIN, new TestTransactionFactory())
```
Comment (elrodro83) [inline tests/integration/src/test/java/org/mule/test/construct/FlowDefaultProcessingStrategyTestCase.java]: why this change?

File: transports/core/src/main/java/org/mule/compatibility/core/DefaultMuleEventEndpointUtils.java [hunk 6]
```diff
@@ -6,10 +6,12 @@
  */
 package org.mule.compatibility.core;
 
+import static org.mule.runtime.core.api.Event.*;
```
Comment (elrodro83) [inline transports/core/src/main/java/org/mule/compatibility/core/DefaultMuleEventEndpointUtils.java]: no *

File: core/src/main/java/org/mule/runtime/core/message/DefaultEventBuilder.java (hunk 1 above)
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/message/DefaultEventBuilder.java]: This wasn't in 3.x, I checked,  seems it found its way in a refactor.  Current behaviour in 4.x relies on the fact that refreshSync does not take I to account reply to handler.

File: transports/core/src/main/java/org/mule/compatibility/core/DefaultMuleEventEndpointUtils.java (hunk 6 above)
Comment (dfeist) [inline transports/core/src/main/java/org/mule/compatibility/core/DefaultMuleEventEndpointUtils.java]: Fixed. Also tests/integration format.

File: tests/integration/src/test/java/org/mule/test/construct/FlowDefaultProcessingStrategyTestCase.java (hunk 4 above)
Comment (dfeist) [inline tests/integration/src/test/java/org/mule/test/construct/FlowDefaultProcessingStrategyTestCase.java]: Ditto

File: tests/integration/src/test/java/org/mule/test/construct/FlowDefaultProcessingStrategyTestCase.java (hunk 5 above)
Comment (dfeist) [inline tests/integration/src/test/java/org/mule/test/construct/FlowDefaultProcessingStrategyTestCase.java]: Ditto.
